src/presetsController.ts

```diff
@@ -734,6 +916,78 @@ export class PresetsController {
+            await this.setTestPreset(preset.defaultTestPreset.name, false/*needToCheckConfigurePreset*/, false/*checkChangingPreset*/);
+        }
+    }
+
+    private async guessPackagePreset(): Promise<void> {
+        const selectedConfigurePreset = this.project.configurePreset?.name;
+        let currentPackagePreset: string | undefined;
+        if (selectedConfigurePreset) {
+            preset.expandConfigurePresetForPresets(this.folderPath, 'package');
+            const packagePresets = preset.allPackagePresets(this.folderPath);
+            for (const packagePreset of packagePresets) {
+                // Set active package preset as the first valid package preset matches the selected configure preset
+                if (packagePreset.configurePreset === selectedConfigurePreset && !packagePreset.hidden) {
+                    await this.setPackagePreset(packagePreset.name, false/*needToCheckConfigurePreset*/, false/*checkChangingPreset*/);
+                    currentPackagePreset = this.project.packagePreset?.name;
+                }
+                if (currentPackagePreset) {
+                    break;
+                }
+            }
+        }
+
+        if (!currentPackagePreset) {
+            // No valid buid preset matches the selected configure preset
+            await this.setPackagePreset(preset.defaultPackagePreset.name, false/*needToCheckConfigurePreset*/, false/*checkChangingPreset*/);
+        }
+    }
+
+    private async guessWorkflowPreset(): Promise<void> {
+        const selectedConfigurePreset = this.project.configurePreset?.name;
+        let currentWorkflowPreset: string | undefined;
+        if (selectedConfigurePreset) {
+            preset.expandConfigurePresetForPresets(this.folderPath, 'workflow');
+            const workflowPresets = preset.allWorkflowPresets(this.folderPath);
+            for (const workflowPreset of workflowPresets) {
+                // Set active workflow preset as the first valid workflow preset (matching the selected configure preset is not a requirement as for the other presets types)
+                if (!workflowPreset.hidden) {
+                    await this.setWorkflowPreset(workflowPreset.name, false/*needToCheckConfigurePreset*/, false/*checkChangingPreset*/);
+                    currentWorkflowPreset = this.project.workflowPreset?.name;
+                }
+                if (currentWorkflowPreset) {
+                    break;
+                }
+            }
+        }
+
+        if (!currentWorkflowPreset) {
+            // No valid workflow preset matches the selected configure preset
+            await this.setWorkflowPreset(preset.defaultWorkflowPreset.name, false/*needToCheckConfigurePreset*/, false/*checkChangingPreset*/);
+        }
+    }
+
     private async checkConfigurePreset(): Promise<preset.ConfigurePreset | null> {
         const selectedConfigurePreset = this.project.configurePreset;
         if (!selectedConfigurePreset) {
```

Comment:
Not your change, but these options need to be "Select Configure Preset" and "Later", see the casing difference.